datus/agent/node/compare_node.py

```
@@ -56,24 +57,35 @@ def update_context(self, workflow: Workflow) -> Dict:
             return {"success": False, "message": f"Comparison context update failed: {str(e)}"}
 
     def _execute_compare(self) -> CompareResult:
-        """Execute comparison analysis between SQL and expectation."""
+        """
+        Execute SQL comparison in a synchronous (non-streaming) mode.
+        """
+        if not isinstance(self.input, CompareInput):
+            raise ValueError("Input must be a CompareInput instance")
```

Ruff (0.14.1)

Prefer `TypeError` exception for invalid type

(TRY004)